### ISSUE TITLE
Unprompted interrupted sync with nexcloud bookmarks on greater amount of bookmarks

### ISSUE THREAD

**Issue by afischer211** [floccusaddon/floccus#1244]
### Which version of floccus are you using?

4.17.0

### Sync method

Nextcloud Bookmarks

### Which browser are you using? In case you are using the phone App, specify the Android or iOS version and device please.

Chromium-based (current version of Vivaldi)

### Which version of Nextcloud Bookmarks are you using? (if relevant)

11.0.1

### Which version of Nextcloud? (if relevant)

24.0.4

### Describe the Bug

I try to sync a bookmark-folder with more than 1000 bookmarks. With firefox it runs successful, with chrome and vivaldi it fails with "E027: Sync process was interrupted"
A second folder (separate floccus account) with less bookmarks syncs successful.

### Expected Behavior

Sync the bookmarks without errors.

### To Reproduce

Create a folder with many bookmarks (more than 1000) and try to sync it (download or merge).

### Debug log provided

- [ ] I have provided a debug log file

**Comment by afischer211** [comment]
Can I configure timeouts or count of retries?`May be the bookmark-app inside Nextcloud is to slow?

**Comment by marcelklehr** [comment]
There is a handler in floccus that detects when the computer goes to sleep. Perhaps that is triggered for you. This only happens when a 2s interval has not been called for 12s, though, which I find unlikely.

**Comment by marcelklehr** [comment]
A timeout of Nextcloud should not cause an interrupted sync error. Interrupted sync generally only happens when you manually cancel a sync run, or the computer has woken up from sleep, or you closed the browser during a sync run.

**Comment by afischer211** [comment]
Hmmh, I run the sync (runs many many minutes until error) - in the log-file I see misc actions and the last line is the message about the aborted sync. So the sync seems to abort himself without my interaction.

**Comment by vitorhmcorreia** [comment]
Hello,

I've recently migrated to floccus using nextcloud 26, syncing on Edge (all of which are running the latest available versions as of this date), and I'm having the same behaviour originally described in the ticket.

I am trying to sync over 100k bookmarks and it's driving me mad :) 
I've tested multiple iterations of settings, as well as chrome and going back and forth from direct nextcloud syncing and webdav.

Since the last post, have there been any developments or, at least, an understanding of what may be happening?

Cheers, thanks!

**Comment by marcelklehr** [comment]
> trying to sync over 100k
> [...]
> an understanding of what may be happening?

That's alot of bookmarks. My guess would be the extension process crashes due to memory consumption.

**Comment by pmorch** [comment]
My experience using it on Android:

Nextcloud Hub 5 (27.0.0)
Nextcloud bookmarks: 13.0.1
Floccus Android App installed today, last update March 24, 2023. I was not able to use the app to see the app version or to find it in Google Play.

I have 4k+ bookmarks and adding a new bookmark looks like it takes a couple of seconds after I hit the "Save" button (which by itself is too long). After that I see the list of bookmarks (without my new bookmark). After another couple of seconds, the sync process starts.

The sync process takes multiple minutes (I'd estimate 5 minutes).

I did not expect this to take multiple minutes to sync a single new bookmark. Downloading an export from nextcloud is virtually instant and in the end I have an export file of about 1MB. 

During and after sync process the app is unresponstive. After multiple minutes , I see 
![E027SyncProcessWasInterrupted](https://github.com/floccusaddon/floccus/assets/159292/7e9aac87-232b-4eb1-8c37-57ea6d3d5da7)

After which I have no choice but to kill the app. If I restart the app, it is very slow. I had no option but to uninstall the app again.

So in short: I was not able to use the app.

**Comment by marcelklehr** [comment]
I've just tested syncing with Nextcloud Bookmarks on Chrome with 88K bookmarks and it works quite well. Takes 1+GB of RAM, though. I haven't tested this in the android app, yet, though.

**Comment by vitorhmcorreia** [comment]
@marcelklehr awesome news 🥳, I'll keep you in the loop with my findings.

**Comment by marcelklehr** [comment]
I can reproduce the problems on Android. It doesn't seem to be an OOM issue. More about responsiveness

**Comment by vitorhmcorreia** [comment]
> @marcelklehr awesome news 🥳, I'll keep you in the loop with my findings.

@marcelklehr happy to report that the latest version of floccus is working brilliantly with my awful amount of bookmarks 😂
truly appreciate your work!

**Comment by marcelklehr** [comment]
Closing as the original issue seems to have been resolved. @pmorch please open a separate issue if this is still happening for you on Android :blue_heart:
